src/pluggy/_manager.py

```diff
@@ -181,7 +192,20 @@ def parse_hookimpl_opts(self, plugin: _Plugin, name: str) -> HookimplOpts | None
         customize how hook implementation are picked up. By default, returns the
         options for items decorated with :class:`HookimplMarker`.
         """
-        method: object = getattr(plugin, name)
+
+        if _attr_is_property(plugin, name):
+            # @property methods can have side effects, and are never hookimpls
+            return None
+
+        method: object
+        try:
+            method = getattr(plugin, name)
+        except AttributeError:
+            # AttributeError: '__signature__' attribute of 'plugin' is class-only
+            # can be raised when trying to access some descriptor/proxied fields
```

Comment:
```suggestion
            # May be raised when trying to access some descriptor/proxied fields
```

Reply:
The reason I suggest to remove it is that the logic handles any AttributeError which may happen for other reasons. The `__signature__` thing is one case. It's not very clear what it means. I think if you can make this error happen in a test it will be better.

Reply:
I did here but it was removed from the PR for being too pydantic-specific: https://gist.github.com/pirate/01ca7a6b41595af9a480

Afaik this case is not triggered in the normal code unless you're checking a few specific weird objects with descriptors like Django/Pydantic models, I don't want to pull in all that complexity.